packages/core/core/src/PackagerRunner.js

```diff
@@ -94,3 +103,41 @@ export default class PackagerRunner {
     return contents;
   }
 }
+
+/*
+ * Build a mapping from async, url dependency ids to web-friendly relative paths
+ * to their bundles. These will be relative to the current bundle if `publicUrl`
+ * is not provided. If `publicUrl` is provided, the paths will be joined to it.
+ * 
+ * These are used to translate any placeholder dependency ids written during
+ * transformation back to a path that can be loaded in a browser (such as
+ * in a "raw" loader or any transformed dependencies referred to by url).
+ */
+function generateDepToBundlePath(bundle: InternalBundle) {
+  let depToBundlePath = new Map<string, FilePath>();
+  bundle.assetGraph.traverse(node => {
+    if (node.type !== 'dependency') {
+      return;
+    }
+
+    let dep = node.value;
+    if (!dep.isURL || !dep.isAsync) {
+      return;
+    }
+
+    let [bundleGroupNode] = bundle.assetGraph.getNodesConnectedFrom(node);
+    invariant(bundleGroupNode && bundleGroupNode.type === 'bundle_group');
+
+    let [entryBundleNode] = bundle.assetGraph.getNodesConnectedFrom(
+      bundleGroupNode
+    );
+    invariant(entryBundleNode && entryBundleNode.type === 'bundle_reference');
+
+    let entryBundle = entryBundleNode.value;
+    depToBundlePath.set(
+      dep.id,
+      urlRelative(nullthrows(bundle.name), nullthrows(entryBundle.name))
```

Comment:
Part of me wants to avoid relative paths all together. I know they're portable, and we should know exactly where they are relative to this bundle, but something feels safer about using a path from the root all the time. For example, do all asset types support relative paths?